app/models/concerns/orchestration/external_ipam.rb

```diff
@@ -0,0 +1,79 @@
+module Orchestration::ExternalIPAM
+  extend ActiveSupport::Concern
+  include Orchestration::Common
+  include SubnetsHelper
+
+  included do
+    after_validation :queue_external_ipam
+    before_destroy :queue_external_ipam_destroy
+  end
+
+  def generate_external_ipam_task_id(action, interface = self)
+    id = [interface.mac, interface.ip, interface.identifier, interface.id].find {|x| x&.present?}
+    "external_ipam_#{action}_#{id}"
+  end
+
+  protected
+
+  def set_external_ip
+    if ip_available
+      subnet.external_ipam_proxy.add_ip_to_subnet(ip, subnet)
+    else
+      self.errors.add :ip, :taken
+      self.errors.add :interfaces, _('Some interfaces are invalid')
```

Comment:
This is a terrible error we do, I would love to get rid of it in the future.

Please report some nice error like:

          self.errors.add :ip, _('External IPAM does not have available IP')